File: sources/academy/webscraping/scraping_basics_python/exercises/wikipedia_country_links.py
import httpx
from bs4 import BeautifulSoup
from urllib.parse import urljoin

listing_url = "https://en.wikipedia.org/wiki/List_of_sovereign_states_and_dependent_territories_in_Africa"
response = httpx.get(listing_url)
response.raise_for_status()

soup = BeautifulSoup(response.text, "html.parser")

for name_cell in soup.select('.wikitable tr td:nth-child(3)'):
    link = name_cell.select_one('a')
    if link and 'href' in link.attrs:
        url = urljoin(listing_url, link['href'])
        print(url)

File: sources/academy/webscraping/scraping_basics_javascript/exercises/wikipedia_country_links.mjs
import * as cheerio from 'cheerio';

const listingUrl = 'https://en.wikipedia.org/wiki/List_of_sovereign_states_and_dependent_territories_in_Africa';
const response = await fetch(listingUrl);

if (!response.ok) {
  throw new Error(`HTTP ${response.status}`);
}

const html = await response.text();
const $ = cheerio.load(html);

for (const element of $('.wikitable tr td:nth-child(3)').toArray()) {
  const $nameCell = $(element);
  const $link = $nameCell.find('a').first();
  if ($link.length) {
    const url = new URL($link.attr('href'), listingUrl).href;
    console.log(url);
  }
}
